server/channels/store/localcachelayer/user_terms_of_service_layer_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package localcachelayer
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/mattermost/mattermost/server/v8/channels/store/storetest"
+	"github.com/mattermost/mattermost/server/v8/channels/store/storetest/mocks"
+)
+
+func TestUserTermsOfServiceStore(t *testing.T) {
+	StoreTest(t, storetest.TestUserTermsOfServiceStore)
+}
+
+func TestUserTermsOfServiceCacheByUser(t *testing.T) {
+	fakeUserTermsOfService := model.UserTermsOfService{UserId: "user123", TermsOfServiceId: "123", CreateAt: 11111}
+	logger := mlog.CreateConsoleTestLogger(t)
+
+	t.Run("first call not cached, second cached and returning same data", func(t *testing.T) {
+		mockStore := getMockStore(t)
+		mockCacheProvider := getMockCacheProvider()
+		cachedStore, err := NewLocalCacheLayer(mockStore, nil, nil, mockCacheProvider, logger)
+		require.NoError(t, err)
+
+		userTermsOfService, err := cachedStore.UserTermsOfService().GetByUser("user123")
+		require.NoError(t, err)
+		assert.Equal(t, userTermsOfService, &fakeUserTermsOfService)
+		mockStore.UserTermsOfService().(*mocks.UserTermsOfServiceStore).AssertNumberOfCalls(t, "GetByUser", 1)
+
+		userTermsOfService, err = cachedStore.UserTermsOfService().GetByUser("user123")
+		require.NoError(t, err)
+		assert.Equal(t, userTermsOfService, &fakeUserTermsOfService)
+		mockStore.UserTermsOfService().(*mocks.UserTermsOfServiceStore).AssertNumberOfCalls(t, "GetByUser", 1)
+	})
```

Comment:
Is this test really checking that the second call returned cached data? Shouldn't we be asserting 0 on the second `AssertNumberOfCalls`?

Reply:
AssertNumberOfCalls(t, "GetByUser", 1) means "assert that the method has been called exactly 1 time total" - not "assert that it was called 1 time since the last check."

Reply:
Oooh, got it! Thanks for the explanation :)